src/circuit.rs

```diff
@@ -212,17 +293,28 @@ impl<'r, F: Field> Region<'r, F> {
         column: Column<Fixed>,
         offset: usize,
         mut to: V,
-    ) -> Result<Cell, Error>
+    ) -> Result<AssignedCell<VR, F>, Error>
     where
         V: FnMut() -> Result<VR, Error> + 'v,
-        VR: Into<Assigned<F>>,
+        for<'vr> Assigned<F>: From<&'vr VR>,
         A: Fn() -> AR,
         AR: Into<String>,
     {
-        self.region
-            .assign_fixed(&|| annotation().into(), column, offset, &mut || {
-                to().map(|v| v.into())
-            })
+        let mut value = None;
+        let cell =
+            self.region
+                .assign_fixed(&|| annotation().into(), column, offset, &mut || {
+                    let v = to()?;
+                    let value_f = (&v).into();
+                    value = Some(v);
```

Comment:
Is the closure guaranteed to be called immediately by `assign_fixed` (if it is called at all)?

Reply:
As @str4d pinned out, the lifetime of the closure is such that it could only be called before `assign_fixed` returns if it is called at all.